CMakeLists.txt

```diff
@@ -121,8 +123,12 @@ include(${CMAKE_CURRENT_SOURCE_DIR}/scripts/build-info.cmake)
 #
 # Compile flags
 #
+if (LLAMA_SYCL)
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
How deep is the C++17 dependency in the SYCL backend?

It's okay to optionally include it like this, but I'm wondering if it is realistic to implement this in C++11 at some point - it would be in better harmony with the rest of the codebase.

Reply:
Actually the icpx compiler expects C++17 standard and SYCL has dependency on that version. We thought about this process having same version C++11 but it causes compilation errors due to dependency on c++17 headers.
@NeoZhangJianyu , @AidanBeltonS and others can add on this.

Reply:
Just to add a bit more info, it is not just that the SYCL compiler, icpx, expects C++17. C++17 is a core aspect within the SYCL open standard. Any SYCL2020 code is expected to be C++17 conformant, so the relationship is deeper than just the specific implementation of the Khronos specification. I would say the dependency between SYCL and C++17 is hard, and it would likely not work well if SYCL specific features were compiled with C++11.

From the spec: https://registry.khronos.org/SYCL/specs/sycl-2020/pdf/sycl-2020.pdf
`The SYCL specification is now based on the core language of C++17, as described in Section 3.9.1. Features of C++17 are now enabled within the specification, such as deduction guides for class template argument deduction`